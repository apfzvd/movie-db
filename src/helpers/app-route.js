import React from 'react'
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom'

import Layout from '../components/layout'

const AppRoute = ({ component: Component, wrapper: Wrapper = Layout, ...rest }) => {
  return (
    <Route
      {...rest}
      exact
      render={matchProps => (
        <Wrapper>
          <Component {...matchProps} />
        </Wrapper>
      )}
    />
  );
}

AppRoute.propTypes = {
  component: PropTypes.node,
  wrapper: PropTypes.node,
}

export default AppRoute
