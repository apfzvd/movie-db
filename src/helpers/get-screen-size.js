const getScreenSize = () => {
  const screenSizes = {
    medium: '(min-width: 30em)',
    large: '(min-width: 60em)',
    xlarge: '(min-width: 80em)',
  }
  const isMedium = window.matchMedia(screenSizes.medium).matches // 480px
  const isLarge = window.matchMedia(screenSizes.large).matches // 960px
  const isXLarge = window.matchMedia(screenSizes.xlarge).matches // 1280px

  if (isXLarge) {
    return 'xlarge';
  }

  if (isLarge) {
    return 'large';
  }

  if (isMedium) {
    return 'medium';
  }

  return 'mobile'
};

export default getScreenSize
