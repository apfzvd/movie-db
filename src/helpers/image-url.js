export default imageId => `https://image.tmdb.org/t/p/w500${imageId}`;
