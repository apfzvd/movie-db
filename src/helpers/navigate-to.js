
import history from '../history'

export default path => history.push(path)
