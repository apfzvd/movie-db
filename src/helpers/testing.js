import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

export const shallowConnected = (component, reducer, initialState) => {
  const store = createStore(reducer, initialState)
  return shallow(<Provider store={store}>{component}</Provider>)
}
