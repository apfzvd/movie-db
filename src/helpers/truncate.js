export default (input, len) => {
  if (input.length > len) {
    return `${input.substring(0, len)}...`
  }
  return input
}
