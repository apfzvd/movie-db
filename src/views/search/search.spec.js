import React from 'react'
import { shallowConnected } from '../../helpers/testing'

import Search from './search-component'
import reducer from '../../components/layout/layout-store'

const props = { match: { params: 'teste' } }

describe('Search component', () => {
  describe('#render', () => {
    it('render correctly', () => {
      const wrapper = shallowConnected(<Search {...props} />, reducer, {
        getState: () => ({ layout: { screenSize: 'small' } }),
      })
      expect(wrapper.debug()).toMatchSnapshot()
    })
  })
})
