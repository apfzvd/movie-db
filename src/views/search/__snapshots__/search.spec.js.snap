// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component #render render correctly 1`] = `
"<ContextProvider value={{...}}>
  <Search match={{...}} />
</ContextProvider>"
`;
